index.js

```diff
@@ -34,9 +38,10 @@ module.exports = function() {
 
       // Unfortunately rework unconditionally splits selectors by comma, so our
       // :matches(...) selector gets munged.
-      var actualSelectors = rule.selectors.join(',').replace(/:matches\(.*\)/g, function(substr) {
+      var actualSelectors = rule.selectors.join(',').replace(/:matches\(.*?\)/g, function(substr) {
```

Comment:
No worries. I very well could be doing it improperly.

My intention was that the regex will now only match up to the first closing paren. However, the `.*` is greedy and will match everything between `:matches(` and the last `)` of a string. So the regex would continue when there was a chained selector all the way to the last closing paren of the joined selectors:

```js
$ node
> var joinedSelector = '.foo:matches(:hover,:active,:focus),.baz:matches(:hover,:active,:focus)'
undefined
> joinedSelector.replace(/:matches\(.*\)/g, function(substr) {
  return substr.replace(/,/g, '&')
}).split(',') 
[ '.foo:matches(:hover&:active&:focus)&.baz:matches(:hover&:active&:focus)' ]
> joinedSelector.replace(/:matches\(.*\)/g, function(substr) {
  console.log(substr)
  return substr.replace(/,/g, '&')
}).split(',')
:matches(:hover,:active,:focus),.baz:matches(:hover,:active,:focus)
[ '.foo:matches(:hover&:active&:focus)&.baz:matches(:hover&:active&:focus)' ]
```

So, the `.*` was the right approach, up until the closing paren because it resulted in an `&` outside the `:matches` parens where a selector was originally grouped.

However, according to the article, I could introduce the minimal matcher, `?`, to make the `.*` "lazy but helpful" [[reference](http://www.rexegg.com/regex-quantifiers.html#helpful)]. This ensures that everything this is matched up until the first closing paren, ensuring that grouped matches are treated separately:

```js
> joinedSelector.replace(/:matches\(.*?\)/g, function(substr) {
  console.log(substr)
  return substr.replace(/,/g, '&')
}).split(',')
:matches(:hover,:active,:focus)
:matches(:hover,:active,:focus)
[ '.foo:matches(:hover&:active&:focus)',
  '.baz:matches(:hover&:active&:focus)' ]
```

With the introduction of the `?` we can ensure that the grouped selectors aren't combined. I'm sure there's likely a more elegant way to achieve what I was intending to do. But I'm anything but a master of regex.